### PR TITLE
add `MSBuildTreatWarningsAsErrors`

### PR DESCRIPTION
**Pull request by rkm**
## Proposed Changes

Treat all warnings as errors. Individual warnings can be disabled if needed when a simple fix isn't available.

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New Feature (non-breaking change which adds functionality)
-   [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation-Only Update (if none of the other choices apply)
    -   In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

-   [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/main/CONTRIBUTING.md) guidelines for this repository
-   [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
-   [ ] Updated any relevant API documentation
-   [ ] Created or updated any tests if relevant
-   [x] Created a [news](https://github.com/SMI/SmiServices/blob/main/news/README.md) file
    -   NOTE: This **_must_** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
-   [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/main/CONTRIBUTORS.md) file 🚀
-   [x] Requested a review by the `@SMI/Reviewers` team

### CODE REVIEW

**Comment by jas88** [comment]
@rkm Mathnet.Numerics workaround PR now open for HicServices/SynthEHR, should be able to unblock this once that merges and releases.

**Comment by jas88** [comment]
@rkm Almost there now - once we pull in the next RDMP release that should all be fixed.

**Comment by rkm** [comment]
Needs #1956

**Review by howff — APPROVED** [review]

**Pull request merged**
